### DISCUSSION TITLE
aao remove mesh in boxでメッシュが消えない

### DISCUSSION THREAD

**Discussion by huryumonji** [anatawa12/AvatarOptimizer#1223] [Q&A]
プレビューでは消えてるのですが、playモードにすると消したはずの部分も表示されてしまいます。
他オブジェクトではうまく出来たのでやり方は間違っていないはずです。原因に思い当たるところはございますか？

又、aao remove mesh in boxは非破壊(やり直しが出来る)ですが、aao remove mesh by maskはテクスチャが上書きされてしまいます。元に戻す方法はありますか？

**Comment by anatawa12** [comment]
> プレビューでは消えてるのですが、playモードにすると消したはずの部分も表示されてしまいます。
> 他オブジェクトではうまく出来たのでやり方は間違っていないはずです。原因に思い当たるところはございますか？

考えられることとしては、Remove Mesh in Box は Skinned Mesh Renderer のいちを基準に箱を指定する仕様ですが、プレビュー等では正しく実装されていない時代があったため、バージョンが古いなどの場合にはSkinned Mesh Rendererを原点に移動させると良い可能性があります。

> aao remove mesh by maskはテクスチャが上書きされてしまいます。元に戻す方法はありますか？

保存するまでは undo ができます。
マスクを保存してウィンドウを閉じたあとにUndoすると自体はできませんが、白く塗りつぶすことで消すと指定していた範囲を消さないように変更することができます。
Brush Colorというところが Mask Texture Editor にあるのでそこで White を選択してください。

**Reply by huryumonji** [reply]
バージョンが古いというのは何のバージョンのことですか？
先程色々いじってみましたが、boxを全く動かさなかった場合オブジェクト全部が消え(プレビューでは半分しか消えてない)、少しでも動かすとオブジェクトが全く消えなくなります。なので位置は関係なさそうです…

**Reply by anatawa12** [reply]
Avatar Optimizerのバージョンです。

うーんSkinnedMeshRendererのスケールは1,1,1ですか？

**Reply by anatawa12** [reply]
最近別のバグ報告により、ボーンの存在しないメッシュでRemove Mesh in Box が正しく動かないことがあるという報告がありました。その削除しようとしているメッシュにはボーンが存在しますでしょうか
